includes/class-llms-dom-document.php

```diff
@@ -0,0 +1,160 @@
+<?php
+/**
+ * A convenient wrapper for the DOMDocument Class
+ *
+ * @package LifterLMS/Classes
+ *
+ * @since [version]
+ * @version [version]
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * LLMS_DOM_Document Class
+ *
+ * @since [version]
+ */
+class LLMS_DOM_Document {
+
+	/**
+	 * Stores the load method name
+	 *
+	 * @var string
+	 */
+	private $load_method = 'load_with_mb_convert_encoding';
+
+	/**
+	 * Stores the HTML string to load
+	 *
+	 * @var string
+	 */
+	private $source;
+
+	/**
+	 * Stores the DOMDocument instance
+	 *
+	 * @var DOMDocument
+	 */
+	private $dom;
+
+	/**
+	 * Stores loading errors
+	 *
+	 * @var null|WP_Error
+	 */
+	private $error;
+
+	/**
+	 * This forces DOMDocument to convert non-utf8 characters into HTML entities and without relying on `mb_convert_encoding()`.
+	 *
+	 * @var string
+	 */
+	private $utf8_fixer = '<meta id="llms-get-dom-doc-utf-fixer" http-equiv="Content-Type" content="text/html; charset=utf-8">';
+
+	/**
+	 * Constructor
+	 *
+	 * @since [version]
+	 *
+	 * @param string $source An HTML string, either a full HTML document or a partial string.
+	 * @return void|WP_Error
+	 */
+	public function __construct( $source ) {
+
+		if ( ! class_exists( 'DOMDocument' ) ) {
+			return new WP_Error( 'llms-dom-document-missing', __( 'DOMDocument not available.', 'lifterlms' ) );
```

Comment:
This return will work to prevent further execution but the error will not actually be returned as the code seems to expect it will. See further comments on the `llms_get_dom_document()` function